### PR TITLE
Update Note Widget

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Update the note widget added in https://github.com/Automattic/simplenote-android/pull/627 with the following:
- add launching configure activity when note not found (i.e. deleted)
- add launching signin activity when not signed in
- add touch feedback on Marshmallow and later
- add analytics tracking
- update configure activity from full-screen to dialog
- update hard-coded values to resources references
- update background color and corner radius
- update views to match app style
- update maximum lines of text
- update loading text and color
- update signin text and color
- update preview image
- update code style
- remove unused import statements
- remove unused resources

The corner radius of the widget was updated from 5dp to 2dp to match the default card view.  The text color and case were updated to differentiate between note content and a note state.  The `simplenote_dark_grey` (`#2D3034`) color and title case are used when the widget is showing note content.  The `simplenote_light_grey` (`#777777`) color and sentence case are used when the widget is in a loading/error state and is not showing content.  A drop shadow was added to the preview image to simulate elevation while placing the widget on the home screen.  The maximum number of lines was updated from 5 to 4 since 5 lines was larger than the widget view and wasn't showing the ellipsis on Android Q.  The configure activity was updated from full-screen to dialog to follow [Material guidelines for widget configure display](https://material.io/design/platform-guidance/android-widget.html).  See the screenshots below for illustration.

widget_note

widget_loading

widget_found

widget_signin

widget_lines

widget_preview

### Test
These steps assume the AOSP launcher (i.e. Nexus/Pixel launcher).  The steps to add the widget to the home screen may be slightly different based on the launcher used.  Other steps should be the same regardless of launcher.
1. Long-press home screen.
2. Tap ***Widgets*** option.
3. Scroll to ***Simplenote*** widget.
4. Notice ***To-Do List*** preview image.
5. Long-press ***Note*** widget.
6. Place ***Note*** widget on home screen.
7. Notice ***Select Note*** dialog with note list.
8. Tap note in list.
9. Notice note title shown in widget.
10. Tap widget.
11. Notice note is shown in app.
12. Go to ***All Notes***.
13. Delete note from Step 8.
14. Go to ***Trash***.
15. Empty trash.
16. Go to home screen.
17. Notice ***Note not fund*** shown in widget.
18. Tap widget.
19. Notice ***Select Note*** dialog with note list.
20. Tap ***Cancel*** button to dismiss dialog.
21. Go to ***Settings***.
22. Tap ***Sign Out*** option.
23. Notice signin screen shown.
24. Go to home screen.
25. Notice ***Sign in to use widget*** shown in widget.
26. Tap widget.
27. Notice signin screen shown.

### Review
@daniloercoli, @maxme, and @nbradbury, I requested you as reviewer since you have contributed to Simplenote in the past.  @bummytime, I requested you as a reviewer to keep you informed of Simplenote changes.  @SylvesterWilmott, I requested you as a reviewer for design.  Only one developer is required to review these changes, but anyone can perform the review.

### CODE REVIEW

File: Simplenote/src/main/java/com/automattic/simplenote/NoteWidgetConfigureActivity.java [hunk 1]
```diff
@@ -152,9 +192,9 @@ public void onClick(View view) {
                     PendingIntent pendingIntent = PendingIntent.getActivity(context, mAppWidgetId, intent, 0);
 
                     // Set widget content
-                    views.setOnClickPendingIntent(R.id.widget_layout, pendingIntent);
-                    views.setTextViewText(R.id.widget_text, note.getTitle());
-                    widgetManager.updateAppWidget(mAppWidgetId, views);
+                    mRemoteViews.setOnClickPendingIntent(R.id.widget_layout, pendingIntent);
+                    mRemoteViews.setTextViewText(R.id.widget_text, note.getTitle());
```
Comment (maxme) [inline Simplenote/src/main/java/com/automattic/simplenote/NoteWidgetConfigureActivity.java]: This `setTextViewText` is called here but the color is not updated. 

When you create a new widget the text color remains `simplenote_light_grey` from line 137 of  `NoteWidget.java`. It's correctly updated after opening the app.

I think this is the only place where that can happen

Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/NoteWidgetConfigureActivity.java]: I added `setTextViewText` to `NoteWidgetConfigureActivity` in 72ae01c.

**Comment by theck13** [comment]
> We should migrate widgets created from previous app versions or do something with them (show an error?).

The widget code in `develop` hasn't actually been released yet.  I wanted to make these updates before we release to avoid exactly that.

**Comment by bummytime** [comment]
@maxme  Since the existing widget code was never released into the wild, are we good to go on this PR then? 😄

File: Simplenote/src/main/java/com/automattic/simplenote/NoteWidgetConfigureActivity.java (hunk 1 above)
Comment (maxme) [reply Simplenote/src/main/java/com/automattic/simplenote/NoteWidgetConfigureActivity.java]: Fix confirmed!

**Comment by maxme** [comment]
LGTM, :shipit:

**Comment by bummytime** [comment]
Thanks @maxme!

**Comment by himat** [comment]
Hi, I downloaded the simple note android app from the play store, but I don't see a widget at all? 
Version 1.8.2

**Comment by theck13** [comment]
Look for the "Note" widget with the "To-Do List" preview image.  See the description in the ***Test*** section above.

**Pull request merged**